packages/aws-cdk-lib/aws-events-targets/lib/redshift-query.ts

```diff
@@ -0,0 +1,134 @@
+import { bindBaseTargetConfig, singletonEventRole } from './util';
+import * as events from '../../aws-events';
+import * as iam from '../../aws-iam';
+import * as secretsmanager from '../../aws-secretsmanager';
+import * as sqs from '../../aws-sqs';
+
+/**
+ * Configuration properties of an Amazon Redshift Query event.
+ */
+export interface RedshiftQueryProps {
+
+  /**
+   * The Amazon Redshift database to run the query against.
+   *
+   * @default - dev
+   */
+  readonly database?: string;
+
+  /**
+   * The Amazon Redshift database user to run the query as. This is required when authenticating via temporary credentials.
+   *
+   * @default - No Database user is specified
+   */
+  readonly dbUser?: string;
```

Comment:
Is a dbUser a Role or some other IAM related resource?

Reply:
This is not an IAM user or Role. It's a redshift user, so it does not have an associated Construct.

Reply:
There is but is apart of an alpha construct: https://docs.aws.amazon.com/cdk/api/v2/docs/@aws-cdk_aws-redshift-alpha.User.html

Reply:
According to the original issue that is linked in this PR, you cannot use an alpha module alongside a matured service. It was recommended in the issue in that case to use string types rather than the redshift module.

In addition, this event target is also used for the serverless module, whereby the serverless module does not have an L2 construct, and no mention of a database user resource.

Reply:
> In addition, this event target is also used for the serverless module

Why is this only scoped to serverless? I don't see anything in the EventBridge Docs that suggest this is only for the RedshiftServerless Resource.

I want to make sure we are not catering to a specific solution here.

On the Alpha part.
I understand this. It's actually very unfortunate because we then can't introduce typing unless we we create more properties. Not a huge fan but that is outside of this change and not something we can directly change right now.

Reply:
This is not only scoped to Redshift serverless, which is why I've used a string type for the database, and a clusterArn for the cluster, rather than specifically targeting either the `Cluster` or `CfnWorkgroup` resources

Reply:
My fault! I miss read the statement. Which makes more sense now. Sorry about the confusion